### PR TITLE
fix(iam): add permission to create log group

### PR DESCRIPTION
**Pull request by svenlito**
## Description
add iam CreateLogGroup permission

## Motivation and Context
To be in line with the `AWSLambdaBasicExecutionRole`.

### CODE REVIEW

File: iam.tf [hunk 1]
```diff
@@ -57,6 +57,7 @@ data "aws_iam_policy_document" "logs" {
     effect = "Allow"
 
     actions = [
+      "logs:CreateLogGroup",
```
Comment (antonbabenko) [inline iam.tf:60]: This one action should be only allowed if `!var.use_existing_cloudwatch_log_group`. Will you be able to update the code?

Comment (svenlito) [reply iam.tf:60]: @antonbabenko I'll have a look at it.

File: iam.tf [hunk 2]
```diff
@@ -56,11 +56,10 @@ data "aws_iam_policy_document" "logs" {
   statement {
     effect = "Allow"
 
-    actions = [
-      "logs:CreateLogGroup",
+    actions = concat([
       "logs:CreateLogStream",
       "logs:PutLogEvents",
-    ]
+    ], compact([!var.use_existing_cloudwatch_log_group ? "logs:CreateLogGroup" : ""]))
```
Comment (svenlito) [inline iam.tf:62]: @antonbabenko not sure if this is ok?

Comment (antonbabenko) [reply iam.tf]: Almost, e9ef635

**Comment by antonbabenko** [comment]
[v1.35.0](https://github.com/terraform-aws-modules/terraform-aws-lambda/releases/v1.35.0) has been just released.

**Pull request merged**